src/server/main_service.cc

```diff
@@ -1326,6 +1325,123 @@ bool StartMultiExec(DbIndex dbid, Transaction* trans, ConnectionState::ExecInfo*
   return true;
 }
 
+struct ExecutionSquasher {
+  static void Execute(absl::Span<StoredCmd> cmds, ConnectionContext* cntx) {
+    ExecutionSquasher es{cmds, cntx};
+    es.Iterate();
+  }
+
+ private:
+  ExecutionSquasher(absl::Span<StoredCmd> cmds, ConnectionContext* cntx)
+      : cmds_{cmds}, cntx_{cntx}, base_cid_{cntx->transaction->GetCId()} {
+  }
+
+  bool CheckSquashing(StoredCmd* cmd) {
```

Comment:
`CheckSquashing` feels like a const operation but it seems it does more than checking